tests/test_views.py

```diff
@@ -49,6 +49,36 @@ def test_get_auth_success(self):
         self.assertEqual(response.status_code, 302)
         self.assertEqual(response.url, '/success')
 
+    @override_settings(LOGIN_REDIRECT_URL='/success')
+    def test_get_auth_success_next_url(self):
+        """Test successful callback request to RP with custom `next` url."""
+        user = User.objects.create_user('example_username')
+        user.is_active = True
+        user.save()
+
+        get_data = {
+            'code': 'example_code',
+            'state': 'example_state'
+        }
+        url = reverse('oidc_authentication_callback')
+        request = self.factory.get(url, get_data)
+        request.session = {
+            'oidc_state': 'example_state',
+            'oidc_login_next': '/foobar'
+        }
+        callback_view = views.OIDCAuthenticationCallbackView.as_view()
+
+        with patch('mozilla_django_oidc.views.auth.authenticate') as mock_auth:
+            with patch('mozilla_django_oidc.views.auth.login') as mock_login:
+                mock_auth.return_value = user
+                response = callback_view(request)
+
+                mock_auth.assert_called_once_with(code='example_code', state='example_state')
+                mock_login.assert_called_once_with(request, user)
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, '/foobar')
```

Comment:
On second thought, you can omit this test altogether and add only the last two assert in `test_get_auth_success` test.